### PR TITLE
Prevent DeleteUser API abuse

### LINKED ISSUE #10119: [API] [Bug] get 500 on Dashborard when delete org
tested on current master: 29151b90c62d4f997a5503c3e0d42468f404a73c
1. Create Org
2. Delete Org via API

-> you get ` routers/user/home.go:57:getDashboardContextUser() [E] GetOrganizations: user does not exist [uid: 3, name: , keyid: 0]`

### PR DESCRIPTION
**Pull request by 6543**
close #10119

### CODE REVIEW

**Comment by 6543** [comment]
I think we should backport this ...

**Review by lafriks — APPROVED** [review]

File: routers/api/v1/admin/user.go [hunk 1]
```diff
@@ -227,6 +228,11 @@ func DeleteUser(ctx *context.APIContext) {
 		return
 	}
 
+	if u.IsOrganization() {
+		ctx.Error(http.StatusUnprocessableEntity, "", fmt.Errorf("%s is an organisation not an user", u.Name))
```
Comment (jolheiser) [inline routers/api/v1/admin/user.go]:
```suggestion
		ctx.Error(http.StatusUnprocessableEntity, "", fmt.Errorf("%s is an organization not a user", u.Name))
```

**Comment by 6543** [comment]
I run in this trap while writing tests for the sdk ...

**Review by davidsvantesson — APPROVED** [review]

File: routers/org/setting.go [hunk 2]
```diff
@@ -115,7 +115,7 @@ func SettingsDeleteAvatar(ctx *context.Context) {
 	ctx.Redirect(ctx.Org.OrgLink + "/settings")
 }
 
-// SettingsDelete response for delete repository
+// SettingsDelete response for delete organisation
```
Comment (jolheiser) [inline routers/org/setting.go]:
```suggestion
// SettingsDelete response for deleting an organization
```

**Review by jolheiser — APPROVED** [review]
Small nits otherwise LGTM

**Comment by lafriks** [comment]
Please send backport

**Comment by 6543** [comment]
done -> #10128

**Comment by lunny** [comment]
I think it's better to move the check into `models.DeleteUser`.

**Comment by 6543** [comment]
should I send a refactor PR?

**Comment by guillep2k** [comment]
> 
> 
> I think it's better to move the check into `models.DeleteUser`.

So orgs can't be deleted ever? 🤔

**Comment by 6543** [comment]
@guillep2k Orgs have there own delete function:
https://github.com/go-gitea/gitea/blob/b3c72a7c4a935e805b87ff2e1911429628ace7c5/models/org.go#L257-L275

**Comment by 6543** [comment]
@lunny ->  #10134

**Pull request merged**